runtime/vm/vmprops.c

```diff
@@ -572,35 +574,45 @@ initializeSystemProperties(J9JavaVM * vm)
 		}
 	}
 
-#if JAVA_SPEC_VERSION < 12
-	/* Following system properties are defined via java.lang.VersionProps.init(systemProperties) and following settings within System.ensureProperties() */
-	{
-		const char *classVersion = NULL;
-		const char *specificationVersion = NULL;
-
-		/* Properties that always exist */
-		switch (j2seVersion) {
-		case J2SE_18:
-			classVersion = "52.0";
-			specificationVersion = "1.8";
-			break;
-		case J2SE_V11:
-		default:
-			classVersion = "55.0";
-			specificationVersion = "11";
-			break;
-		}
-		rc = addSystemProperty(vm, "java.class.version", classVersion, 0);
-		if (J9SYSPROP_ERROR_NONE != rc) {
+	if (JAVA_SPEC_VERSION == 8) {
+		specificationVersion = "1.8";
+	} else {
+		bufLen = j9str_printf(PORTLIB, NULL, 0, "%d", JAVA_SPEC_VERSION);
```

Comment:
I'd like to avoid the allocation here if we can.  

Can you look at C preprocessor stringification - doing something like:
```C
#define STRINGIFY(x) #x 
specificationVersion = STRINGIFY(JAVA_SPEC_VERSION)

```

Reply:
Yeah, trying the hints from Pete's comment about `J9_STR(x)`.